flycheck.el

```diff
@@ -7744,7 +7744,12 @@ Otherwise return the previously used cache directory."
 Look up the directory hierarchy from FILE for a directory
 containing a file that matches REGEXP."
   (locate-dominating-file
-   file
+   ;; We must not pass regular file here because
+   ;; `locate-dominating-file' will just pass it into it's second
+   ;; lambdaargument . But our lambda expects only directories.
+   (if (file-directory-p file)
```

Comment:
Yes, but if I pass `"/usr/bin"` there then calling `file-name-diroctory` on it will produce `"/usr/"` directory, thus unconditionally stripping `bin` while I may have meant to consider too.